pkg/provisioner/ironic/ironic_test.go

```diff
@@ -20,58 +26,113 @@ func init() {
 	logf.SetLogger(logf.ZapLogger(true))
 }
 
-func TestGetUpdateOptsForNodeWithRootHints(t *testing.T) {
-	rotational := true
-	host := &metal3v1alpha1.BareMetalHost{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "myhost",
-			Namespace: "myns",
-			UID:       "27720611-e5d1-45d3-ba3a-222dcfaa4ca2",
+func TestProvisionerIsReady(t *testing.T) {
+
+	cases := []struct {
+		name      string
+		ironic    *mockServer
+		inspector *mockServer
+
+		expectedIronicCalls    string
+		expectedInspectorCalls string
+		expectedIsReady        bool
+		expectedRequeueAfter   time.Duration
+		expectedError          string
+	}{
+		{
+			name:      "IsReady",
+			ironic:    newMockServer(6385).addDrivers(),
+			inspector: newMockServer(5050),
```

Comment:
I think there's an impediment here: https://github.com/metal3-io/baremetal-operator/blob/231ba343d896ff6ae120c99ef22ebfdde1da52ee/pkg/provisioner/ironic/ironic.go#L50-L73. The `init()` method gets triggered before the test, and thus the related (not accessible) package variables fetching the env vars are already set once the ` newProvisioner` call is made (and I think it's the reason why those env vars are set in the Makefile and also in hack/unit.sh). It could help to refactor the configuration initialization, maybe with a lazy setup.

Reply:
I'm not really convinced that a unit test should be making network calls at all.